syntax = "proto3";
package windows_service;

service Service{
    rpc get_info (ServiceInfoRequest) returns (stream ServiceResponse) {};
    rpc start (StartServiceRequest)  returns (ServiceResponse) {};
    rpc stop (StopServiceRequest)  returns (ServiceResponse) {};
    rpc restart (RestartServiceRequest)  returns (ServiceResponse) {};
}

message BasicRequest{
    string computer = 1;
    optional string name = 2;
}

message ServiceResponse{
    string computer = 1;
    string name = 2;
    optional string description = 3;
    optional bool accept_pause = 4;
    optional bool accept_stop = 5;
    optional string caption = 6;
    optional uint32 check_point = 7;
    optional string creation_classname = 8;
    optional string delayed_auto_start = 9;
    optional string desktop_interact = 10;
    optional string display_name = 11;
    optional string error_control = 12;
    optional uint32 exit_code = 13;    
    optional string path_name = 14;
    optional uint32 process_id = 15;
    optional uint32 service_specific_exit_code = 16;    
    optional string service_type = 17;
    optional bool started = 18;
    optional string start_mode = 19;
    optional string state = 20;
    optional string status = 21;
    optional string system_creation_classname = 22;
    optional string system_name = 23;
    optional string start_name = 24;
    optional uint32 tag_id = 25;
    optional uint32 wait_hint = 26;
    optional string details_request_datetime = 27;
}

message ServiceInfoRequest{
    BasicRequest request = 1;
}

message StartServiceRequest{
    BasicRequest request = 1;
}

message StopServiceRequest{
    BasicRequest request = 1;
}

message RestartServiceRequest{
    BasicRequest request = 1;
}